### PR TITLE
Deliverable 3

### PR DESCRIPTION
**Pull request by angelmartinez-wz**

### CODE REVIEW

File: usecase/pokemon.go [hunk 1]
```diff
@@ -129,3 +131,81 @@ func (u PokemonUsecase) DeletePokemon(id string) (model.Pokemon, error) {
...
+
+	itemsInt, _ := strconv.Atoi(items)
+	itemsPerWorkerInt, err := strconv.Atoi(itemsPerWorker)
+	poolSize := utils.GetIntFromDivision(itemsInt, itemsPerWorkerInt)
+	limit := utils.GetIntFromDivision(len(pokemonList), poolSize)
+
+	fmt.Println(poolSize)
+	fmt.Println(limit)
+
+	if err == nil {
```
Comment (FerVT) [inline usecase/pokemon.go]: You can avoid this indentation if you do 
```
if err != nil{
    err = errors.New("cannot get Pokemons")
    return pokemonList, err
}
```

<your logic here>

File: usecase/pokemon.go [hunk 2]
```diff
@@ -129,3 +131,81 @@ func (u PokemonUsecase) DeletePokemon(id string) (model.Pokemon, error) {
...
+
+		var wg sync.WaitGroup
+		wg.Add(poolSize)
+
+		for i := 0; i < poolSize; i++ {
+			go func(i int) {
+				tmpPokemon := pokemonList[i]
+				if tmpPokemon.Id%2 != 0 {
+					chOd <- tmpPokemon.Id
+				} else if tmpPokemon.Id%2 != 1 {
```
Comment (FerVT) [inline usecase/pokemon.go]: I don't think you need to add this second if.
If the number is not odd then it is even, no need to double check it

File: usecase/pokemon.go [hunk 3]
```diff
@@ -129,3 +131,81 @@ func (u PokemonUsecase) DeletePokemon(id string) (model.Pokemon, error) {
...
+				tmpPokemon := pokemonList[i]
+				fmt.Println("odd", i, ":", tmpPokemon)
+				oddPokemonList = append(oddPokemonList, tmpPokemon)
+				wg.Done()
+			}
+		}()
+
+		go func() {
+			defer close(chEv)
+			for i := range chEv {
```
Comment (FerVT) [inline usecase/pokemon.go]: I think you actually need a way to get out of this for loop, if not you may be getting into a deadlock. Maybe after each iteration you could check if a shutdown channel has been closed, if so you can break out of the loop

File: utils/utils.go [hunk 4]
```diff
@@ -61,4 +63,11 @@ func GetPokemonByKey(params url.Values, pokemonList model.PokemonList) model.Pok
...
 
 	return filteredPokemon
-}
\ No newline at end of file
+}
+
+/*
+GetIntFromDivision returns an int from a division of two ints
+*/
+func GetIntFromDivision(x int, y int) int {
```
Comment (FerVT) [inline utils/utils.go]: Remember that you can make this line a bit shorter by grouping common data types -> `(x, y int)`

File: usecase/pokemon.go (hunk 1 above)
Comment (angelmartinez-wz) [reply usecase/pokemon.go]: 👌

File: usecase/pokemon.go (hunk 2 above)
Comment (angelmartinez-wz) [reply usecase/pokemon.go]: My mistake, I forgot to delete this unnecessary validation, thanks!

File: utils/utils.go (hunk 4 above)
Comment (angelmartinez-wz) [reply utils/utils.go]: 💯

File: usecase/pokemon.go (hunk 3 above)
Comment (angelmartinez-wz) [reply usecase/pokemon.go]: Refactored with the example

**Pull request merged**